analyzer/tests/Makefile

```diff
@@ -66,7 +66,7 @@ test_functional_in_env: venv_dev
 		$(ACTIVATE_DEV_VENV) && $(FUNCTIONAL_TEST_CMD)
 
 test_build_logger:
-	$(MAKE) -C tools/build-logger test
+	$(MAKE) -C tools/build-logger -j1 test
```

Comment:
The problem with the `test` target is that it depends on the `all` target:
https://github.com/Ericsson/codechecker/blob/f11eb1a09c796a083a8fb9ea7d94946a262fe8da/analyzer/tools/build-logger/tests/Makefile#L8

And what `all` target  will do is to build the package:
https://github.com/Ericsson/codechecker/blob/f11eb1a09c796a083a8fb9ea7d94946a262fe8da/analyzer/tools/build-logger/Makefile#L76

This is why your make command failes and gives you an error when you run it on multiple threads.

Can't we remove this dependency from the `test` target and assume you already built a package from ldlogger? That would solve the race condition.

Reply:
Possible, but why should we assume it's already done?

Reply:
If we do not assume that and if you run the for example the `all` target with multiple threads it is not guaranteed that the dependencies will be run in the correct order. For example the `all` target depends on the `pack32bit` target:
https://github.com/Ericsson/codechecker/blob/f11eb1a09c796a083a8fb9ea7d94946a262fe8da/analyzer/tools/build-logger/Makefile#L78

This target depends on two other targets: `32bit` and `packbin`. The order is important here because the `packbin` target will assume that `32bit` already run. If you run it with multiple threads this is not guaranteed.

Reply:
These should not depend on the order. If that's the case then `packbin` should explicitly depend on `32bit`.

Reply:
Anyway, I updated the patch to simply omit the `all` target from the `test` target deps.